### ISSUE TITLE
LLM2CLIP: Powerful Language Model Unlocks Richer Visual Representation

### ISSUE THREAD

**Issue by bigshanedogg** [bigshanedogg/survey#43]
## Problem statement

1. CLIP의 text encoder를 LLM으로 교체하여 LLM의 open world knowledge와 advanced text understanding을 leverage할 수 있는가? 
  - long and complex captions
  - InternVL 프레임워크에서 LLM의 차용은 있었지만, 이것은 retraining을 동반하는 비용비효율적인 방법이다.
2. LLM은 Constrastive learning에 적합한 representation을 반환하는가?  - CLIP은 text encoder를 from the scratch로 학습하기 떄문에 이러한 우려가 없었지만, LLM을 동일하게 처음부터 학습시킬 수는 없다.

## Glossary
- SimCSE 방식의 Contrastive learning
  - ![Image](https://github.com/user-attachments/assets/033a1aed-d875-4ce8-b5fb-b42f1393039a) 

## Data details

- 어떤 데이터를 실험에 이용했는지

### Caption contractive (CC) fine-tuning
<img width="488" alt="Image" src="https://github.com/user-attachments/assets/bfe4accb-975b-41ca-b162-b244c62d6ac1" />
<img width="515" alt="Image" src="https://github.com/user-attachments/assets/fab8ea39-6274-4d9e-aa0c-81422f4697bc" />

  - SLM 때부터 있던 약점이지만, LLM도 여전히 서로 다른 텍스트에도 유사한 output feature를 반환한다. 실제로 Finetuning 전에는 LLM은 t2t retrieval 성능이 매우 좋지 않다. -> low discriminative power
  - LLM의 output space를 조정하기 위해 CC3M에 대해 LoRA로 SimCSE와 같은 방식으로 튜닝한다. (positive pairs는 ShareCaptioner로 reparaphrasing된 캡션을 사용)
  - LLM의 causal masking을 bidirectional attention으로 변경 후 MNTP (Masked Next Token Prediction)으로 이 bidirectional attention에 대한 initialization 부여
    - MNPT는 그 자체로는 큰 gain이 없었고, bidirectional attention이 약간의 gain을 가져왔다. 대부분의 gain은 SimCSE 방식의 학습에서 왔다.

###  Vision Encoder fine-tuning
<img width="499" alt="Image" src="https://github.com/user-attachments/assets/71afcd89-bc04-4411-a0f0-fc3bd59bfb62" />

  - LLM은 frozen한 상태로 CLIP 학습하여, LLM의 open world knowledge가 사라지는 것을 막고, 학습 코스트를 절감함. 
  - 단, adapter와 projector 학습. adapter는 linear layers로 이루어져있으며, projector는 1개의 linear layer. 다른 works에서는 통상 묶어서 projector라고도 하는데, 나눈 것으로 보임.
  - 이렇게 학습할 경우, original CLIP의 fine-tuning과 유사한 정도의 학습 코스트가 소요된다.
    - CC 학습에는 8A100을 사용했으며, Finetuning에는 8H100으로 9시간이 걸렸다고 함.

### 적용된 방법론이 효과와 더불어, 얼마나 generalization이 잘되는지
<img width="794" alt="Image" src="https://github.com/user-attachments/assets/64c75964-a15f-40da-838d-ac7a79d79f8a" />

    - CLIP의 스케일 뿐 아니라, EVA02 등의 CLIP variants에도 동일한 경향성이 적용된다.
    - CC3M 대신에 CC15M, CC60M 등으로 학습량을 늘렸을 때도 증가하는 경향을 보인다.
  - ShareGPT4V, Urban-1k, DOCCI 등은 long text retrieval 데이터셋인데, 앞의 short text 뿐 아니라, long text에서도 높은 증가를 보인다.
<img width="927" alt="Image" src="https://github.com/user-attachments/assets/418cd66a-3951-4c8f-a068-6f9c9158487b" />

  - 이러한 개선은 다른 LLM을 사용할 때도 같은 경향성을 보인다.

### Cross lingual 평가
 <img width="999" alt="Image" src="https://github.com/user-attachments/assets/c5cd2e19-6ecb-4f07-af39-ee5fe20e71e2" />

  - Cross lingual 세팅에서도 LLM의 지식을 활용하는 것으로 보인다. 중국어에 finetuned된 CLIP보다 더 높은 성능을 보인다. (LLM2CLIP의 학습 시에 중국어 데이터에 대해 노출시키지 않았음에도)

### 이미지넷 zero-shot 평가
<img width="515" alt="Image" src="https://github.com/user-attachments/assets/6eea6646-0fd7-4cc0-87b3-05c7e8f64f6e" />

  - 이미지넷의 zero-shot 퍼포먼스는 오히려 감소한다. 이는 LLM2CLIP의 학습 과정에서 alignment 페이즈가 짧기도 하고, 이미지넷의 클래스들이 word frequency가 낮은 단어들이기 때문인 것으로 추정한다.
    - linear probe 후에는 오히려 성능이 증가하는 것이 이를 뒷받침한다.

### Vision Encoder의 VLLM 적용 평가
<img width="1023" alt="Image" src="https://github.com/user-attachments/assets/0faf81e3-223b-4e5b-a5e0-b7e6f789d1ff" />

  - LLM2CLIP으로 학습한 Vision Encoder는 VLLM으로 사용할 때도 improvement를 보인다.

## Conclusion
- text encoder를 바로 LLM으로 교체하는 것은 오히려 좋지 않다.
  - LLM은 discriminability가 부족하기 때문에 이를 개선하는 것이 CLIP 적용에 key modification이다.
- LLM을 CLIP에 적용하는 것은 long text, built-in-knowledge의 활용에 있어 강점을 부여한다.
  - LLM은 우선 기본적으로 context length가 길기 때문에 구조적인 문제도 없고, 위의 long text 태스크들은 복잡한 추론을 요구하는데, 이에 대해서도 성능 향상을 보인다.
- scalability가 적용된다.

## Limitations

- 제안된 구조의 약점
- 저자가 숨기고 싶어하는 부분 혹은 논리적 허점
